### ISSUE TITLE
iOS crash, mangled name mismatch between framework(that links lottie as dynamic library) and hosting app

### ISSUE THREAD

**Issue by DrAma999** [airbnb/lottie-ios#1292]
## Check these before submitting:
- [x] The issue doesn't involve an [Unsupported Feature](https://github.com/airbnb/lottie-ios/blob/master/README.md#supported-after-effects-features)
- [x] This issue isn't related to another open issue

## This issue is a:
- [x] Crashing Bug
- [x] Regression (Something that once worked, but doesn't work anymore)

## Which Version of Lottie are you using?

Lottie 3.1.9 with XCode 12.2

## What Platform are you on? 

- [x] iOS

## What Language are you in?
- [x] Swift
- [x] Objective-C

## Expected Behavior 
Animation should be loaded as with the previous lottie versions

## Actual Behavior
Our framework integrate lottie as a dynamic library, we updated the lottie version to 3.1.9 and shipped the framework.
Ou client noted that a crash is triggered right after calling a method that instantiate the lottie animation view.
Lottie in the framework is only linked dynamically as dependency.
We are both using latest version of cocoa pods and Xcode 12.2

After trying to to instantiate the Lottie Animation view we get a crash on referenced symbol.
**dyld: Symbol not found: _$s6Lottie0A18BackgroundBehaviorO15pauseAndRestoreyA2CmFWC
  Referenced from: /private/var/containers/Bundle/Application/DE33AC8A-3C7B-413B-8E28-A894C0492487/YYY.app/Frameworks/XXX.framework/XXX
  Expected in: /private/var/containers/Bundle/Application/DE33AC8A-3C7B-413B-8E28-A894C0492487/YYY.app/Frameworks/Lottie.framework/Lottie
 in /private/var/containers/Bundle/Application/DE33AC8A-3C7B-413B-8E28-A894C0492487/YYY.app/Frameworks/XXX.framework/XXX**
Same happens on another symbol even if we remove that particular part of code on another symbol.
It seems that there is a mismatch on mangled names between what has been used in our framework and what is downloaded from our client. 
That didn't happen using 3.1.8 on Xcode 12.1

## Code Example
```
let animationView = AnimationView()
```

**Comment by TLizer** [comment]
@DrAma999 Have you tried removing Xcode derived data? Had similar problem recently after bumping one of the frameworks and it happens that Xcode was caching some symbols in there.

**Comment by DrAma999** [comment]
Hello @TLizer Yep, clean install, removed derived data directory, removed cocoa pods cache, removed everything :-)
Tried on 3 different mac to avoid caching of something.
But I'm starting to think that this a problem of Xcode or cocoa pods, on an italian swift group there are several developers with a similar issue on other dependency. I'm wandering if this release was compiled for Xcode 12.2, even if coacopods rebuild, maybe there is till something to set up. With the older version of Xcode 12.1 and 3.1.8 i had no issue.

**Comment by DrAma999** [comment]
It seems that also another user had that issue https://stackoverflow.com/questions/65001773/xcframework-dyld-cache-load-error-shared-cache-file-open-failed on Lottie

**Comment by TLizer** [comment]
@DrAma999 
I tried to reproduce this issue and found possible issue. 
I created empty project and embedded Lottie 3.1.8. Then I removed it and added 3.1.9 and reproduced this issue.
It seems that it was caused by search paths that was referencing to both 3.1.8 and 3.1.9 -> Have a look there :)

**Comment by DrAma999** [comment]
@TLizer thank you for your suggestion, I will look into that. At the moment we managed the issue by creating in the host app a post_install in cocoa pods that sets the **Build libraries for distribution** flag to **YES**. Basically we forced module stability. Fun fact: it seems that pointing directly to master doesn't show that issue, probably because is already compatible with Xcode 12.2?
On the other hand I was reading that a lot of user are suffering from that issue with other major libraries, such as _Alamofire_, _RXSwift_ etc, and the same solution works also for them. The only things in common are Cocoapods and Xcode 12.2.
